File: src/main/java/org/technbolts/util/HtmlUtils.java
package org.technbolts.util;

import java.io.PrintWriter;
import java.io.StringReader;
import java.io.StringWriter;

import org.apache.commons.lang.StringEscapeUtils;
import org.w3c.tidy.Tidy;

public class HtmlUtils {

	public static String htmlToText(String html) {
		StringWriter out = new StringWriter();
		StringWriter err = new StringWriter();
		Tidy tidy = new Tidy();
		tidy.setErrout(new PrintWriter(err));
		tidy.setPrintBodyOnly(true);
		tidy.setMakeClean(true);
		tidy.setHideComments(true);
		tidy.setNumEntities(false);
		tidy.setAsciiChars(true);
		//Node node = 
		tidy.parse(new StringReader(html), out);
		String str = out.toString();
		str = str.replaceAll("<[^>]+>", " ");
		str = StringEscapeUtils.unescapeHtml(str);
		return str;
	}
}

File: src/main/scala/org/technbolts/util/HtmlUtils.scala
package org.technbolts.util

import org.apache.commons.lang.StringEscapeUtils
import org.w3c.tidy.Tidy
import java.io.{StringWriter, PrintWriter, StringReader}

/**
 * 
 */
object HtmlUtils {

  def htmlToText(html: String): String = {
    var out: StringWriter = new StringWriter
    var err: StringWriter = new StringWriter
    var tidy: Tidy = new Tidy
    tidy.setErrout(new PrintWriter(err))
    tidy.setPrintBodyOnly(true)
    tidy.setMakeClean(true)
    tidy.setHideComments(true)
    tidy.setNumEntities(false)
    tidy.setAsciiChars(true)
    tidy.parse(new StringReader(html), out)

    var str: String = out.toString
    // remove html elements
    str = str.replaceAll("<[^>]+>", " ")
    str = StringEscapeUtils.unescapeHtml(str)
    return str
  }
}